lib/model/coordinate.dart

```diff
@@ -5,21 +5,47 @@ import 'package:google_maps_flutter/google_maps_flutter.dart';
 class Coordinate {
   final double _lat;
   final double _lng;
-  final String _floor;
-  final String _building;
-  final String _campus;
+  String _floor;
+  String _building;
+  String _campus;
+
+  //attributes to be used for google places search results
+  String _gPlaceTitle;
+  //String _gPlacePhotos;
+  String _gPlaceAddress;
+  String _gPlacePhone;
+  String _gPlaceWebsite;
+
   String _type;
   Set<Coordinate> _adjCoordinates = HashSet<Coordinate>();
 
-  Coordinate(this._lat, this._lng, this._floor, this._building, this._campus,
-      {type, adjCoordinates}) {
-    _type = type;
+ // Constructor:
+// All the attributes inside curly bracket are optional/dynamic attributes
+  Coordinate(
+    this._lat,
+    this._lng,
+    this._floor,
+    this._building,
+    this._campus, {
+    type,
+    adjCoordinates,
+    gPlaceTitle,
+    //String  gPlacePhoto,
+    gPlaceAddress,
+    gPlacePhone,
+    gPlaceWebsite,
+  }) {
+        _type = type;
     if (adjCoordinates != null) {
       _adjCoordinates = adjCoordinates;
       for (var adjCoordinate in _adjCoordinates) {
         adjCoordinate.addAdjCoordinate(this);
       }
     }
+     _gPlaceTitle = gPlaceTitle;
```

Comment:
The formatting is a bit off, make sure you use the flutter format on every document you worked on before commiting/PR

Reply:
Thanks for your comment, I made required changes. :-)